src/main/java/com/google/sps/servlets/AboutServlet.java

```diff
@@ -32,29 +33,41 @@
 
 @WebServlet(urlPatterns = "/about")
 public class AboutServlet extends HttpServlet {
+  private String staticResponse;
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    System.out.println("REQUEST AT: " + request.getServletPath());
-    response.setContentType("text/html;");
-
+  public void init() {
     JinjavaConfig config = new JinjavaConfig();
     Jinjava jinjava = new Jinjava(config);
     try {
       jinjava.setResourceLocator(
           new FileLocator(
               new File(this.getClass().getResource(ResourceConstants.TEMPLATES).toURI())));
-    } catch (URISyntaxException e) {
+    } catch (URISyntaxException | FileNotFoundException e) {
       System.err.println("templates dir not found!");
     }
 
     Map<String, Object> context = new HashMap<>();
-    context.put("url", "/");
-    String template =
-        Resources.toString(
-            this.getClass().getResource(ResourceConstants.TEMPLATE_ABOUT), Charsets.UTF_8);
-    String renderedTemplate = jinjava.render(template, context);
+    context.put("url", "/about");
+
+    try {
+      String template =
+          Resources.toString(
+              this.getClass().getResource(ResourceConstants.TEMPLATE_ABOUT), Charsets.UTF_8);
+      staticResponse = jinjava.render(template, context);
+    } catch (IOException e) {
+      System.err.println("template not found");
+    }
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+
+    if (staticResponse == null) {
+      init();
+    }
 
-    response.getWriter().println(renderedTemplate);
+    response.getWriter().println(staticResponse);
```

Comment:
What if you get an IOException in init()? You will eat it up there and now staticResponse will continue to be null despite calling init().